CONTRIBUTING.md

```diff
@@ -0,0 +1,108 @@
+- Include a test that isolates the bug and verifies that it was fixed.
+
+### New Features
+
+- If you'd like to add a feature to the library that doesn't already exist, feel free to describe the feature in a new [GitHub issue].
+- You can also join us on [Discord]() to discuss some initials thoughts.
+- If you'd like to implement the new feature, please wait for feedback from the project maintainers before spending too much time writing the code. In some cases, enhancements may not align well with the project future development direction.
+- If applicable, implement the code for the new feature and please read the [How To](#how-to-contribute).
+
+### Documentation & Miscellaneous
+
+- If you have suggestions for improvements to the documentation or examples (or something else), we would love to hear about it.
+- As always first file a [Github issue].
+- Implement the changes to the documentation, please read the [How To](#how-to-contribute).
+
+## How To Contribute
+
+### Requirements
+
+- Linux, Mac OS X, or Windows.
+- [Git](https://git-scm.com) (used for source version control).
+- An IDE such as [Android Studio](https://developer.android.com/studio) or [Visual Studio Code](https://code.visualstudio.com/).
+
+### Forking & cloning the repository
+
+- Ensure all the dependencies described in the previous section are installed.
+- Fork `https://github.com/invertase/melos` into your own GitHub account. If
+  you already have a fork, and are now installing a development environment on
+  a new machine, make sure you've updated your fork so that you don't use stale
+  configuration options from long ago.
+- `git clone git@github.com:<your_name_here>/melos.git`
+- `git remote add upstream git@github.com:invertase/melos.git` (So that you
+  fetch from the main repository, not your clone, when running `git fetch` or `git pull`
+
+## Environment Setup
+
+### Local development setup
+
+To setup and use this melos mono repo locally for the purposes of contributing, clone it and run the following commands from the root of the repository:
+
+```bash
+# Install melos if it's not already installed:
+dart pub global activate melos
+
+# Activate 'melos' from path:
+melos activate
+
+# Confirm you now using a local development version:
+melos --help
+# You should now see a banner printed at the top of the help output similar to:
+# ---------------------------------------------------------
+# | You are running a local development version of melos. |
+# ---------------------------------------------------------
+```
+
+### Install
+
+```bash
+dart pub global activate melos
+```
+
+## Performing changes
+
+- Create a new local branch from `main` (e.g. `git checkout -b my-new-feature`)
+- Make your changes (try to split them up with one PR per feature/fix).
+- When committing your changes, make sure that each commit message is clear
+ (e.g. `git commit -m 'Added CONTRIBUTING.md'`).
```

Comment:
The example commit message should follow the conventional commit spec, e.g. `docs: Added CONTRIBUTING.md`.